Feature/test-cloud espresso support

krukow opened the pull request:
A first take for Espresso support for Test beacon

@kmiecikt please review this draft proposal -- I'm aware of quite a bit of duplication, but I wanted to have you own how you want code to be shared

Note: also created a `./mobile-center` helper for OS X which I found useful as a beginner

Note: requires https://github.com/xamarin/test-cloud-frontend/commit/44015bd66a8838c24dd11a2c8a1c9a5d2163dd58 to test this end to end (this should be deployed tomorrow at least to staging)

kmiecikt commented on src/commands/test/lib/espresso-preparer.ts:
```diff
@@ -0,0 +1,149 @@
+    await this.testApkPath();      
+  }
+
+  private async testApkPath(): Promise<string> {
```
The "test prepare uitest" requires "--app-path" option that points to the application file. Should we do the same here?

krukow replied:
yeah `--app-path` is definitely required, and yes, I think we should allow (like uitest tool dir option) the user to override the espresso tests apk file with an option. Thanks for that.

krukow commented:
@kmiecikt addressed your review comments - please re-review.

Successfully uploaded espresso test:

https://mobile.azure.com/users/kakrukow-qggr/apps/squadgoals/test/runs/e78e8a0d-6da3-4617-8949-1cc8472bb0e0

https://testcloud.xamarin.com/test/e78e8a0d-6da3-4617-8949-1cc8472bb0e0/pipeline 


Commands: 
with  `--test-apk-path`
```
./mobile-center test run espresso --app "kakrukow-qggr/squadgoals" \ 
     --devices 108b848e  \ 
    --app-path ~/code/squad-goals-android/app/build/outputs/apk/app-debug.apk \
    --test-series "master" --locale "en_US" \ 
    --test-apk-path ~/code/squad-goals-android/app/build/outputs/apk/app-debug-androidTest.apk
```
with  `--build-dir`
```
./mobile-center test run espresso --app "kakrukow-qggr/squadgoals" \ 
     --devices 108b848e  \ 
    --app-path ~/code/squad-goals-android/app/build/outputs/apk/app-debug.apk \
    --test-series "master" --locale "en_US" \ 
    --build-dir ~/code/squad-goals-android/app/build/outputs/apk
```

christav commented on mobile-center:
```diff
@@ -0,0 +1,2 @@
+#!/bin/bash
```
It's been removed and replaced by the auto-generated npm launch script. I'd rather remove the ad-hoc launch scripts from the repo. For dev purposes we can take the hit to run node/ts-node directly, but it's less confusing for our customers, and easier for us to maintain installation/package publishing, if we have one canonical way to launch the thing.

krukow replied:
removed

christav commented:
@krukow What's the status of this PR? Is it intended to be merged now, or is it still a sample/prototype?

krukow commented:
@christav thanks. I was waiting for a final signoff from @kmiecikt but it's ready to go from my perspective.
